### DISCUSSION TITLE
Using Belay  with multiple devices

### DISCUSSION THREAD

**Discussion by TimAidley** [BrianPugh/belay#146] [Q&A] [answered]
What would be the best way to use belay with multiple devices? The way it is set up at the moment, with the @device decorator being specific to a particular connected device, it doesn't seem like it lends itself very well to multiple devices.

**Comment by BrianPugh** [comment, marked as answer]
The best way would be to use the class-based interface (which is very similar). See the [multiple device example.](https://github.com/BrianPugh/belay/blob/main/examples/08_device_subclassing/main_multiple_devices.py). Each device can have it's own class if they are performing different operations.

**Comment by TimAidley** [comment]
Ah! Perfect. Thank you.
